app/models/grda_warehouse/tasks/scrub_pii/scrub_client_pii_task.rb

```diff
@@ -0,0 +1,115 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module GrdaWarehouse::Tasks::ScrubPii
+  # * Selectively scrub personally identifiable information (PII) for HMIS Clients
+  # * Delete client-related custom hmis data
+  # * Delete versions
+  class ScrubClientPiiTask
+    def self.perform(...)
+      new.perform(...)
+    end
+
+    def perform(client_ids: nil, data_source_ids: nil, custom_scrubber: nil, progress: false)
```

Comment:
Can you say more about the intended use-case for this task?
The original ask was around cleaning up an environment containing PII, without the need to really retain anything except client IDs. This all looks good to me for that purpose.

Since this accepts `client_ids`, should I be reviewing this as if we were to use it to remove PII for an individual client upon request?(Or if certain length of time has passed since enrollment, etc). These are use-cases we have discussed in the past, but haven't been totally spec'd out afaik. If that's the intended use for this then I may have some different feedback, for example clearing the content of the notes rather than fully deleting all the notes.

Reply:
And vice versa, for the _first_ use case (environment cleanup) it would be fine to just drop _all_ the custom data elements, notes, etc.- we're retaining more than we really need for that use case.

Reply:
> Since this accepts client_ids, should I be reviewing this as if we were to use it to remove PII for an individual client upon request?

Yes, that's the intention of this task. It's trying to selectively delete PII from certain client records; the other task is intended to comprehensively scrub all PII from the entire database. I'll add some better inline comments to make that clear.

> I may have some different feedback, for example clearing the content of the notes rather than fully deleting all the notes.

The original spec had "Free-form text (delete) - case notes, custom fields?"  which I took as delete. I think generally sanitizing fields rather than deleting is better so I'm happy to make that change. Two points:
* Some fields probably can't be nulled out since a case-node or cde with a null value could break the app. Safe to replace with a static or faker-value
* Perhaps we shouldn't try to select the subset of sensitive fields from CDE's since we might miss some. We could instead sanitize all CDEs. Maybe we could check-in on this briefly

Reply:
As discussed last time, this is a foundation for scrubbing PII, we will need to revisit the specifics if/when we introduce any features for scrubbing individual client data. As discussed there are several potential use-cases that may have different requirements (eg "anonymize me, but keep my data intact for continued service" vs "remove my data entirely, but keep my data intact for reporting" ).